tachyon/zk/plonk/vanishing/vanishing_utils.h

```diff
@@ -0,0 +1,166 @@
+// polynomial of the 2ᵏ size domain.
+template <typename F, typename Domain, typename ExtendedDomain,
+          typename ExtendedEvals>
+ExtendedEvals DivideByVanishingPoly(const ExtendedEvals& evals,
+                                    const ExtendedDomain* extended_domain,
+                                    const Domain* domain) {
+  ExtendedEvals ret = evals;
+  DivideByVanishingPolyInPlace<F>(ret, extended_domain, domain);
+  return ret;
+}
+
+// Given a |poly| of coefficients  [a₀, a₁, a₂, ...], this returns
+// [a₀, ζa₁, ζ²a₂, a₃, ζa₄, ζ²a₅, a₆, ...], where ζ is a cube root of unity in
+// the multiplicative subgroup with order (p - 1), i.e. ζ³ = 1.
+//
+// |into_coset| should be set to true when moving into the coset, and false
+// when moving out. This toggles the choice of ζ.
+template <typename F, typename ExtendedPoly>
+void DistributePowersZeta(ExtendedPoly& poly, bool into_coset) {
+  F zeta_inv = GetHalo2ZetaInv<F>();
+  F zeta = zeta_inv.Square();
+  std::vector<F> coset_powers{into_coset ? zeta : zeta_inv,
+                              into_coset ? zeta_inv : zeta};
+
+  std::vector<F> coeffs = poly.coefficients().coefficients();
+  base::Parallelize(coeffs,
+                    [&coset_powers](absl::Span<F> chunk, size_t chunk_idx,
+                                    size_t chunk_size) {
+                      size_t i = chunk_idx * chunk_size;
+                      for (F& a : chunk) {
+                        // Distribute powers to move into/from coset
+                        size_t j = i % (coset_powers.size() + 1);
+                        if (j != 0) {
+                          a *= coset_powers[j - 1];
+                        }
+                        ++i;
+                      }
+                    });
+}
+
+// This takes us from the extended evaluation domain and gets us the quotient
+// polynomial coefficients.
+//
+// This function will panic if the provided vector is not the correct length.
+template <typename F, typename ExtendedPoly, typename ExtendedEvals,
+          typename ExtendedDomain>
+ExtendedPoly ExtendedToCoeff(const ExtendedEvals& evals,
+                             const ExtendedDomain* extended_domain) {
+  CHECK_EQ(evals.NumElements(), extended_domain->size());
+
+  ExtendedPoly poly = extended_domain->IFFT(evals);
+
+  // Distribute powers to move from coset; opposite from the
+  // transformation we performed earlier.
+  DistributePowersZeta<F>(poly, false);
+
+  return poly;
+}
+
+// template <typename Poly, typename ExtendedPoly>
+// std::vector<Poly> SplitPolys(const ExtendedPoly& poly) {
+//   std::vector<Poly> polys;
+//   polys.reserve(poly.coefficients().coefficients().size() / 2);
+//   for (size_t i = 0; i < poly.coefficients().coefficients().size(); i += 2) {
+//     polys.emplace_back(
+//         Poly::FromCoefficients({poly.coefficients().coefficients()[i],
+//                                 poly.coefficients().coefficients()[i + 1]}));
+//   }
+//   return polys;
+// }
```

Comment:
What's this?

Reply:
It was a temporary code, but since it's unused, I'll delete it.